tests/test_notify_slack.py

```diff
@@ -0,0 +1,129 @@
+from json import dumps
+from logging import getLogger
+from os import environ
+from unittest.mock import MagicMock, patch
+
+from pytest_subtests import SubTests
+
+from backend.api_keys import EVENT_KEY
+from backend.notify_slack.task import (
+    EVENT_DETAIL_KEY,
+    EVENT_MESSAGE_KEY,
+    EVENT_RECORDS_KEY,
+    EVENT_SNS_KEY,
+    SLACK_CHANNEL_NAME,
+    SLACK_MESSAGE_CHANNEL_KEY,
+    SLACK_MESSAGE_TEXT_KEY,
+    SLACK_NOTIFY_URL_ENV_NAME,
+    STEP_FUNCTION_ARN_KEY,
+    lambda_handler,
+)
+from backend.step_function import Outcome
+from backend.step_function_keys import (
+    ASSET_UPLOAD_KEY,
+    ERRORS_KEY,
+    METADATA_UPLOAD_KEY,
+    STATUS_KEY,
+    STEP_FUNCTION_KEY,
+    VALIDATION_KEY,
+)
+
+from .aws_utils import any_arn_formatted_string, any_lambda_context
+from .general_generators import any_https_url
+
+LOGGER = getLogger("backend.notify_slack.task")
+
+
+@patch("backend.notify_slack.task.http.request")
+@patch("backend.notify_slack.task.get_import_status_given_arn")
+def should_notify_slack_when_url_set(
+    get_import_status_mock: MagicMock, http_mock: MagicMock, subtests: SubTests
+) -> None:
+    # Given
+
+    get_import_status_mock.return_value = get_import_status_output = {
```

Comment:
I don't think the contents of the `get_import_status_given_arn` response is relevant to these tests, because `lambda_handler` simply returns the response verbatim. So I think you could just do `get_import_status_mock.return_value = get_import_status_output = {}`.